dcp-0009/dcp-0009.mediawiki

```diff
@@ -0,0 +1,301 @@
+only to revocation transactions and NOT vote transactions.
+
+When revoking a ticket, the original ticket contribution amounts must first be
+evenly distributed to each output of the revocation transaction.
+
+After the contributions are distributed evenly, there may be a remainder of
+<code>1</code> to <code>numOutputs - 1</code> atoms.  Each atom in the remainder
+must be added to an output index that is selected in a deterministic uniformly
+pseudorandom manner, as described below.
+
+The pseudorandom selection requires a constant value that is known to all full
+nodes.  This constant is derived from the hex representation of the mathematical
+constant Pi (𝜋) and acts as a publicly verifiable Nothing-Up-My-Sleeve number.
+The constant that is used is <code>0x243F6A8885A308D3</code>.
+
+For a revocation being included into a block at height <code>N+1</code>, the
+remainder should be added to outputs as follows:
+# Initialize variables to be used in the deterministic pseudorandom number generation as follows:
+## Construct the initial 32-byte <code>seed</code> as follows:
+### Concatenate the serialized raw bytes of the header of block <code>N</code> with the constant value <code>0x243F6A8885A308D3</code>
+### Hash the result from the previous step with <code>BLAKE-256</code>
+## Initialize a hash: <code>hash = seed</code>
+## Initialize a hash iterator index: <code>hashIterIdx = 0</code>
+## Initialize a hash offset: <code>hashOffset = 0</code>
+## Initialize <code>upperBound</code> to the number of outputs in the revocation transaction
+##: NOTE: The maximum allowed number of outputs for a revocation transaction, and therefore the maximum <code>upperBound</code>, is <code>64</code>
+# For each atom in the remainder:
+## Select a uniformly random integer <code>i</code> in the range <code>[0, upperBound)</code> as follows:
+### While <code>r < 2^32 % upperBound</code>
+#### Select a random 32-bit integer, <code>r</code>, as follows:
+##### Take a 4-byte slice of the <code>hash</code> between positions <code>hashOffset</code> and <code>hashOffset+4</code>: <code>hash[hashOffset:hashOffset+4]</code>
+##### Assign the big endian 32-bit unsigned integer representation of the result of the previous step to <code>r</code>
+##### Increment <code>hashOffset</code> by <code>1</code>
+##### If <code>hashOffset > 7</code>, then "rollover" the hash as follows:
+###### Set <code>hash</code> to the <code>BLAKE-256</code> hash of the <code>seed</code> concatenated with the <code>hashIterIdx</code>
+###### Set <code>hashOffset = 0</code>
+###### Increment <code>hashIterIdx</code> by <code>1</code>
+### Calculate <code>i = r % upperBound</code> to get the uniformly random integer <code>i</code> in the range <code>[0, upperBound)</code>
+## Increment the output at index <code>i</code> by <code>1</code>
+
+===Skipping Script Validation for Revocation Transactions===
+
+This proposal requires that script validation MUST be skipped for version
```

Comment:
I agree, it is probably unnecessary for the most part.  I removed this from most of the specification sections, though I left the preamble for the ticket redeemer output amounts calculation section since otherwise, I feel it's not really obvious what the remainder is.

Reply:
Agree that the ticket redeemer section either requires the explanation there or in the motivation since it is not specified how it works anywhere else.  Updates look good.